hydra_plugins/hydra_wandb_sweeper/_impl.py

```diff
@@ -0,0 +1,856 @@
+                        )
+                    except Exception as e:
+                        if run_status[run.id] == RunStatus.STOPPED:
+                            ret = "Run has been stopped by sweep controller"
+                            status = JobStatus.COMPLETED
+                        else:
+                            ret = e
+                            status = JobStatus.FAILED
+                        run_results.append(
+                            {
+                                "run_id": run.id,
+                                "name": run.name,
+                                "return_value": ret,
+                                "status": status,
+                            }
+                        )
+                    # Catching any system exits from the task function so that we could inspect why the exit happened
+                    # and pass a JobStatus and other useful info to the launcher pertaining to the exit. i.e.,
+                    # don't stop everything just because a task function exited, just passively send the info to the
+                    # launcher.
+                    except SystemExit as e:
+                        ret = e
+                        if e.code == 0 or e.code is None:
+                            status = JobStatus.COMPLETED
+                        else:
+                            status = JobStatus.FAILED
+                        run_results.append(
+                            {
+                                "run_id": run.id,
+                                "name": run.name,
+                                "return_value": ret,
+                                "status": status,
+                            }
+                        )
+                    finally:
+                        if isinstance(ret, BaseException):
+                            ret = repr(ret)
+                        logger.info(
+                            f"Agent finished executing task function under Run {run.id} ({run.name}) "
+                            f"with status: {status} and return value: {ret}"
+                        )
+
+            def agent_run(agent):
+                agent._heartbeat_thread = Thread(target=agent._heartbeat)
+                agent._heartbeat_thread.daemon = True
+                agent._heartbeat_thread.start()
+                agent._run_jobs_from_queue()
+
+            try:
+                agent_status = JobStatus.COMPLETED
+                agent_error = None
+                num_runs_so_far = sum(self.num_runs.values())
+                if num_runs_so_far <= self.max_num_runs:
+                    agent = Agent(
+                        self.wandb_sweeper.sweep_id, count=self.agent_run_count
+                    )
+                    try:
+                        agent._setup()
+                    except Exception as e:
+                        agent_id = None
+                        agent_error = "Sweep has been killed."
```

Comment:
I think there can be other reasons as well, not just killed. For example, IIRC, if an user stops / pauses a sweep, and tries to create an agent without resuming the sweep, the agent throws an error.

Reply:
Ah, fair point. So perhaps a better message is "Sweep is not available"?

Reply:
How about extracting the message from the exception (using `str(e)`) and using that instead?

Reply:
Hmmm, that will make it difficult to figure out if a sweep is available or not when the next batch of agents are to be launched, [since that error message is what's used to figure out if a sweep is available](https://github.com/captain-pool/hydra-wandb-sweeper/blob/c7211452e7b1a50924aa768cb315206f1eb29447/hydra_plugins/hydra_wandb_sweeper/_impl.py#L654-L664). Don't get me wrong, I agree with what you're saying, I'm just trying to figure out another way of detecting whether the sweep is available or not, given the sweep id, before launching the next batch of agents.

Reply:
@captain-pool do you happen to know of a programmatic way of querying whether a sweep is available or not given a sweep id?

Reply:
Apologies for the delay. I checked in with the WandB sweeps team. As of now, there’s currently no way of doing that. Although you might find a `state` property in the documentation for sweeps, which for some reason is non-existent in the code. I will raise an internal ticket to notify the docs team, and raise a feature request with the Sweeps team.

Reply:
Ah darn. Thanks for raising the feature request :)